src/Controller/KeycloakController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Domain;
+use App\Entity\User;
+use App\Handler\UserAuthenticationHandler;
+use App\Repository\DomainRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class KeycloakController extends AbstractController
+{
+    private readonly DomainRepository $domainRepository;
+    private readonly UserRepository $userRepository;
+
+    public function __construct(private readonly EntityManagerInterface $manager, private readonly UserAuthenticationHandler $handler) {
+        $this->domainRepository = $this->manager->getRepository(Domain::class);
+        $this->userRepository = $this->manager->getRepository(User::class);
+    }
+
+    #[Route(path: '/api/keycloak', name: 'api_keycloak_index', methods: ['GET'], stateless: true)]
+    public function index(Request $request): Response
+    {
+        if (null === $domain = $this->domainRepository->findByName($request->query->get('domain') ?? '')) {
+            return $this->json([
+                'message' => 'domain not found',
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        if (null === $search = $request->query->get('search')) {
+            $search = '';
+        }
+
+        if (null === $max = $request->query->get('max')) {
+            $max = 10;
+        }
+
+        if (null === $first = $request->query->get('first')) {
+            $first = 0;
+        }
+
+        $users = $this->userRepository->findUsersByString($domain, $search, $max, $first)->map(function (User $user) {
+            return [
+                'id' => explode('@', $user->getEmail())[0],
+                'email' => $user->getEmail(),
+            ];
+        });
+        return $this->json($users);
+    }
+
+    #[Route(path: '/api/keycloak/count', name: 'api_keycloak_count', methods: ['GET'], stateless: true)]
+    public function count(Request $request): Response
+    {
+        if (null === $domain = $this->domainRepository->findByName($request->query->get('domain') ?? '')) {
+            return $this->json([
+                'message' => 'domain not found',
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        return $this->json($this->userRepository->countDomainUsers($domain));
+    }
+
+    #[Route(path: '/api/keycloak/user/{email}', name: 'api_keycloak_user', methods: ['GET'], stateless: true)]
+    public function get(Request $request, string $email): Response
```

Comment:
```suggestion
    public function getUser(Request $request, string $email): Response
```

Reply:
`getUser` is already defined by parent class `AbstractController`. Any idea for another name, @louis?

Reply:
I called it `getOneUser()` now.